src/audio_decoder.cpp

```diff
@@ -105,19 +104,21 @@ class WMAUnsupportedFormatDecoder : public AudioDecoder {
 		error_message = std::string("WMA audio files are not supported. Reinstall the\n") +
 			"game and don't convert them when asked by Windows!\n";
 	}
-	bool Open(FILE*) override { return false; }
+	bool Open(Filesystem::InputStream) override { return false; }
 	bool IsFinished() const override { return true; }
 	void GetFormat(int&, Format&, int&) const override {}
+	bool Seek(std::streamoff, std::ios_base::seekdir) override { return false; }
 private:
 	int FillBuffer(uint8_t*, int) override { return -1; };
 };
 const char wma_magic[] = { (char)0x30, (char)0x26, (char)0xB2, (char)0x75 };
 
-std::unique_ptr<AudioDecoder> AudioDecoder::Create(FILE* file, const std::string& filename, bool resample) {
+std::unique_ptr<AudioDecoder> AudioDecoder::Create(Filesystem::InputStream stream, const std::string& filename, bool resample) {
 	char magic[4] = { 0 };
-	if (fread(magic, 4, 1, file) != 1)
+	if (stream->read(magic, sizeof(magic)).gcount() == 0) {
```

Comment:
should it be `!= sizeof(magic)` what if I tried to read a 1, 2 or 3 byte file?

Reply:
added now such a ReadIntoObj, just not using it alot